### PR TITLE
setting up directory for tbe_batch_processor c

### LINKED ISSUE #50: C_read Directory of Files
**Description:** 
Develop functionality to process an entire directory of TBE files in C, loading and parsing each file sequentially. This will facilitate batch processing, allowing users to aggregate metadata and handle errors efficiently across multiple files.

**Developer Notes:**
- Implement a function that iterates through all files in a specified directory, selecting only TBE files for processing.
- The function should return parsed data in a structured format (e.g., an array of structs) that is easily accessible and usable for further operations.
- Provide options for generating summary reports by aggregating key metadata fields across all processed TBE files.

**Acceptance Criteria:**
- [x] Functionality to read all TBE files in a directory
- [x]  metadata aggregation
- [x]  skip each non-TBE files and log a warning
- [x]  provide logging or progress updates
- [x]  an option to return summary metadata in a JSON or visual format

### PR DESCRIPTION
**Pull request by BaloneyBoy97**
This PR fixes #50 

python_tbe/c_processing/tbe_batch_processor.c:

   Process individual TBE files.
   Process all TBE files in a directory.
   Aggregate metadata across files (processed/skipped records).
   Generate a metadata summary and write it to a JSON file.

Key Features:

  Uses json-c library to handle JSON generation.
  Aggregates the following metadata:
      Total files processed.
      Total skipped files.
      Total records.
      Average records per processed file.

python_tbe/python_conv/c_tbe_integration.py:

Purpose:

  Provides Python integration for the C-based TBE batch processor.
  Allows easy invocation of the C functions from Python.

Key Features:

  Calls process_tbe_directory to process TBE files in a directory.
  Calls generate_metadata_summary to output JSON and visual summaries.
  Handles errors like invalid directories or missing libraries.

Build Command:

gcc -shared -o python_tbe/c_processing/tbe_batch_processor.so \
python_tbe/c_processing/tbe_batch_processor.c \
-I/opt/homebrew/Cellar/json-c/0.18/include/json-c \
-L/opt/homebrew/Cellar/json-c/0.18/lib -ljson-c

Run Command:

python3 python/src/functions/c_tbe_integration.py c/src/functions/tbe_batch_processor.so ./sample_data

Screenshot 2024-11-29 at 2 22 25 AM

Screenshot 2024-11-29 at 2 23 57 AM

### CODE REVIEW

File: .vscode/c_cpp_properties.json
Comment (mohamdlog) [inline .vscode/c_cpp_properties.json:1]: .vscode should be in .gitignore

File: metadata_summary.json
Comment (mohamdlog) [inline metadata_summary.json:1]: Add to .gitignore.

Comment (BaloneyBoy97) [reply metadata_summary.json:1]: created .gitignore

File: .DS_Store
Comment (mohamdlog) [inline .DS_Store:1]: Needs to be removed from PR.

File: .vscode/c_cpp_properties.json
Comment (mohamdlog) [inline .vscode/c_cpp_properties.json:1]: Needs to be removed from PR

File: metadata_summary.json
Comment (mohamdlog) [reply metadata_summary.json:1]: Needs to be removed from PR

File: python_tbe_archive/.DS_Store
Comment (mohamdlog) [inline python_tbe_archive/.DS_Store:1]: Needs to be removed from PR.

**Pull request closed without merging**